src/effects/VST3/VST3Utils.h

```diff
@@ -52,7 +56,73 @@ class VST3Utils final
    //a human-readable form.
    static wxString MakeAutomationParameterKey(const Steinberg::Vst::ParameterInfo& info);
 
-   //Attempts to extract a unique VST3 parameter id from the key.
-   //Returns true on success.
-   static bool ParseAutomationParameterKey(const wxString& key, Steinberg::Vst::ParamID& paramId);
 };
+
+struct EffectSettings;
+
+struct VST3EffectSettings
+{
+   std::unordered_map<Steinberg::Vst::ParamID, Steinberg::Vst::ParamValue> mValues;
+};
+
+struct VST3Wrapper
+{
+   // For the time being, here we have only members that are needed
+   // to iterate parameters and extract preset state
+   //
+   Steinberg::IPtr<Steinberg::Vst::IEditController> mEditController;
+   Steinberg::IPtr<Steinberg::Vst::IComponent>      mEffectComponent;
+
+   using ParameterInfo = Steinberg::Vst::ParameterInfo;
+
+   using ParameterVisitor = std::function< bool(const ParameterInfo& pi) >;
+
+   // made mutable because it is cached once on demand
+   mutable std::vector<ParameterInfo> mParameterInfos;
```

Comment:
Here and elsewhere, I suggest you learn the conventions for Doxygen comments.  See : https://www.doxygen.nl/manual/docblocks.html

Simply adding `!` after `//` means a comment will also be extracted to the automatically generated web pages.  If you install the program, you can run `doxygen nographs.dox` to generate web pages on your own computer, then open `dox/html/index.html` to test how it looks.  I say this as a recommendation, not a requirement for merging this.

Reply:
Maybe this mutable member should become private.

Reply:
ok, done